### DISCUSSION TITLE
VTI Interface with Dynamic Peer

### DISCUSSION THREAD

**Discussion by xrpixer** [strongswan/strongswan#605] [Q&A]
Looking for understanding and possibly examples if available.

I'd like to have a site to site VPN using VTI interfaces and a dynamic peers.
Basically I would have a hub and spoke, where the hub has multiple dynamic peers configured. The hub IP would be static, but all the clients would have dynamic addresses, either because they're clients getting moved around, they're behind NAT, or their ISP isn't static.

I've got this working without VTI interfaces, but i'd really like the option of doing BGP and routing across the tunnels. I was able to do this as well using GRE and VyOS, following this as a guide -
https://support.vyos.io/en/kb/articles/gre-over-ipsec-for-secure-tunneling-2

But I feel that's really dirty and unwieldy once you get to having ALOT of dynamic peers. It would be great to be able to cut out the GRE part.

Is this currently possible using StrongSwan?
If not, could we consider this for a Feature Request?

I've been able to do this sort of thing using a Juniper SRX, but a StrongSwan box would be much preferred as it'd be a significantly cheaper option.

**Comment by tobiasbrunner** [comment]
Did you read [RouteBasedVPN](https://wiki.strongswan.org/projects/strongswan/wiki/RouteBasedVPN)?
